spark/v3.3/spark-extensions/src/main/antlr/org.apache.spark.sql.catalyst.parser.extensions/IcebergSqlExtensions.g4

```diff
@@ -82,6 +82,13 @@ snapshotRetentionClause
     | WITH SNAPSHOT RETENTION numSnapshots SNAPSHOTS snapshotRetain snapshotRetainTimeUnit
     ;
 
+createReplaceBranchClause
+    : (CREATE | REPLACE) BRANCH identifier branchOptions
+    ;
+
+branchOptions
+    : (AS OF VERSION snapshotId)? (RETAIN snapshotRefRetain snapshotRefRetainTimeUnit)? (snapshotRetentionClause)?;
```

Comment:
I had a comment about simplifying the variables used in https://github.com/apache/iceberg/pull/6637#discussion_r1083330622, could we also follow that here?

Reply:
Done!